### PR TITLE
by default idrac(USB-NIC) interface configure with link local address

### PR DESCRIPTION
**Pull request by prabhakarpujeri**
assigning link local address to idrac  interface. which will help idrac user directly browse idrac web page  using https://idrac.local

### CODE REVIEW

**Comment by poettering** [comment]
What is idrac?

**Comment by poettering** [comment]
Hmm, so is this supposed to be "Dell DRAC"? https://en.wikipedia.org/wiki/Dell_DRAC

The name appears to be configure via a hwdb entry we ship, hence it indeed makes sense to also ship something that enables ipv4ll on the interface by default... @teg, @kaysievers any idea how that should look like?

We could just ship a .network file as suggested, but I am a bit concerned that we might end up shipping too many of those eventually then. Hence: should we support some kind of .network-auto-generated-from-hwdb stuff for cases like this? i have the suspicion this is not going to be the only case like this...

**Comment by ssahani** [comment]
it's Remote Access Controller (iDRAC). These are server management tool for Dell systems. We can access it via http/remote console.

**Comment by prabhakarpujeri** [comment]
@poettering

generating .network file from hwdb is  good idea because we may end up with many .network files.

**Comment by poettering** [comment]
OK, so here's my proposal, after thinking a bit about this. I really don't want to make the udev db or hwdb a configuration store. But I also don't want to carry default .network files for specific hw devices in .network files. To find a middle ground one option might be:
- we introduce a new generic udev property called LINK_PURPOSE= or so, with a possible value LINK_PURPOSE=autoip or so.
- we use the hwdb to set this property for the idrac stuff
- we introduce a new [Match] field called LinkPurpose= or so, that checks for this property
- we ship a default .network file that does IPv4LL+DHCP with a match section of LinkPurpose=autoip

That way networkd isn't used for match specific hw, but hwdb does that. And hwdb is not used to store configuraiton, as networkd does that.

Maybe LinkPurpose= is a shitty name, but you get the gist of it.

@teg, does that make any sense?

File: network/80-idrac-nic.network [hunk 1]
```diff
@@ -0,0 +1,14 @@
+g  This file is part of systemd.
```
Comment (sujithshankar) [inline network/80-idrac-nic.network:1]: This line should begin with '#' not 'g'. 
This is a typing mistake.

**Comment by poettering** [comment]
I guess we can close this and continue discussion in #2150.

**Pull request closed without merging**